src/app/d2-vendors/VendorItemComponent.tsx

```diff
@@ -97,8 +102,16 @@ export default class VendorItemComponent extends React.Component<Props, {}> {
     } else {
       console.log(item);
 
+      const reviewData = trackerService.getD2ReviewDataCache().getRatingData(undefined, item.itemHash);
+
+      if (reviewData && !reviewData.reviews) {
+        const reviewsData = await trackerService.getItemReviewAsync(item.itemHash);
```

Comment:
This is another place that could block for a while - I ended up awaiting the data for the buckets but that's a local operation that should return instantly - this will block opening the popup on updating the review data.